File: leetcode.com/0717_1_bit_and_2_bit_characters/solution.go
func isOneBitCharacter(bits []int) bool {
    found := false
    
    for i := 0 ; i < len(bits) ; i++ {
        if bits[i] == 1 {
            i += 1
        } else if (i + 1 == len(bits)) {
            found = true
        }
    }
    
    return found
}

File: leetcode.com/0717_1_bit_and_2_bit_characters/solution.c
bool isOneBitCharacter(int* bits, int bitsSize)
{
    int pos = 0;
    bool found = false;
    
    while (pos < bitsSize) {
        if (bits[pos+0] == 1 && (bits[pos+1] == 1 || bits[pos+1] == 0) && pos + 2 <= bitsSize) {
            pos += 2;
        } else {
            pos += 1;
            
            if (pos >= bitsSize)
                found = true;
        }
    }
    
    return found;
}

